### DISCUSSION TITLE
How can I populate suggestion items dynamically?

### DISCUSSION THREAD

**Discussion by plstreatmewell** [ueberdosis/tiptap#765] [General Questions] [answered]
Hi there! TipTap is great WYSIWYG Editor. Thanks for developing and maintaining it!

BTW, I have a question.
Currently, the example code populates suggestion items inside Mention initialization.
But I want to dynamically add new items to suggestion items after Editor initialization.
How can I do it?
 
```javascript
data () {  
  return {
    editor: new Editor ({
      extensions: [
        new Mention({
          item: () => [
            { id: 1, name: 'A' },
            { id: 2, name: 'B' },
            { id: 3, name: 'C' },
            { id: 4, name: 'D' }
          ],
          // ........
        })
      ]
    }) 
  }
}
```

Thank you for reading!

**Comment by plstreatmewell** [comment, marked as answer]
-----[SOLVED]-----
Assigning reference of this.items to items inside new Mention() works!
After that, you can push or delete items from this.items and that changes suggestion items too!
```
data () {  
  return {
    items: [
      { id: 1, name: 'A' },
      { id: 2, name: 'B' },
      { id: 3, name: 'C' },
      { id: 4, name: 'D' }
    ],
    editor: new Editor ({
      extensions: [
        new Mention({
          item: this.items,
          // ........
        })
      ]
    }) 
  }
}
```

**Reply by hanspagel** [reply]
Thanks for sharing!
